substrate/frame/balances/src/impl_currency.rs

```diff
@@ -918,3 +918,15 @@ where
 		Self::update_locks(who, &locks[..]);
 	}
 }
+
+impl<T: Config<I>, I: 'static> InspectLockableCurrency<T::AccountId> for Pallet<T, I> {
+	fn balance_locked(id: LockIdentifier, who: &T::AccountId) -> Self::Balance {
+		Self::locks(who)
+			.into_iter()
+			.filter(|l| l.id == id)
+			.collect::<Vec<_>>()
+			.pop()
+			.map(|l| l.amount)
```

Comment:
could use reduce? It also (unsafely) assumes there is always one lock with an ID.

Reply:
I refactored it to use fold instead, much simpler.